// unocss-preset-weapp/dist/rules.cjs
'use strict';

const _default = require('./shared/unocss-preset-weapp.e84c6ce3.cjs');
const filters = require('./shared/unocss-preset-weapp.640c88d4.cjs');
require('./shared/unocss-preset-weapp.d6f74a4e.cjs');
require('@unocss/core');
require('unplugin-transform-class/utils');
require('@unocss/rule-utils');

const questionMark = [
  [
    /^(where|\?)$/,
    (_, { constructCSS, generator }) => {
      if (generator.userConfig.envMode === "dev")
        return `@keyframes __un_qm{0%{box-shadow:inset 4px 4px #ff1e90, inset -4px -4px #ff1e90}100%{box-shadow:inset 8px 8px #3399ff, inset -8px -8px #3399ff}}
${constructCSS({ animation: "__un_qm 0.5s ease-in-out alternate infinite" })}`;
    }
  ]
];

exports.alignments = _default.alignments;
exports.animations = _default.animations;
exports.appearance = _default.appearance;
exports.aspectRatio = _default.aspectRatio;
exports.backgroundStyles = _default.backgroundStyles;
exports.bgColors = _default.bgColors;
exports.borderStyles = _default.borderStyles;
exports.borders = _default.borders;
exports.boxSizing = _default.boxSizing;
exports.colorScheme = _default.colorScheme;
exports.containerParent = _default.containerParent;
exports.cssProperty = _default.cssProperty;
exports.cssVariables = _default.cssVariables;
exports.divides = _default.divides;
exports.flex = _default.flex;
exports.flexGridJustifiesAlignments = _default.flexGridJustifiesAlignments;
exports.floats = _default.floats;
exports.fonts = _default.fonts;
exports.gaps = _default.gaps;
exports.grids = _default.grids;
exports.handlerBorderStyle = _default.handlerBorderStyle;
exports.insets = _default.insets;
exports.justifies = _default.justifies;
exports.lineClamps = _default.lineClamps;
exports.margins = _default.margins;
exports.opacity = _default.opacity;
exports.orders = _default.orders;
exports.outline = _default.outline;
exports.overflows = _default.overflows;
exports.paddings = _default.paddings;
exports.placements = _default.placements;
exports.positions = _default.positions;
exports.rules = _default.rules;
exports.safeArea = _default.safeArea;
exports.sizes = _default.sizes;
exports.spaces = _default.spaces;
exports.svgUtilities = _default.svgUtilities;
exports.tabSizes = _default.tabSizes;
exports.textAligns = _default.textAligns;
exports.textDecorations = _default.textDecorations;
exports.textIndents = _default.textIndents;
exports.textShadows = _default.textShadows;
exports.textStrokes = _default.textStrokes;
exports.transitions = _default.transitions;
exports.verticalAligns = _default.verticalAligns;
exports.willChange = _default.willChange;
exports.zIndexes = _default.zIndexes;
exports.appearances = filters.appearances;
exports.backdropFilterBase = filters.backdropFilterBase;
exports.backgroundBlendModes = filters.backgroundBlendModes;
exports.boxShadows = filters.boxShadows;
exports.boxShadowsBase = filters.boxShadowsBase;
exports.breaks = filters.breaks;
exports.contains = filters.contains;
exports.contentVisibility = filters.contentVisibility;
exports.contents = filters.contents;
exports.cursors = filters.cursors;
exports.displays = filters.displays;
exports.dynamicViewportHeight = filters.dynamicViewportHeight;
exports.filterBase = filters.filterBase;
exports.filters = filters.filters;
exports.fontSmoothings = filters.fontSmoothings;
exports.fontStyles = filters.fontStyles;
exports.hyphens = filters.hyphens;
exports.isolations = filters.isolations;
exports.mixBlendModes = filters.mixBlendModes;
exports.objectPositions = filters.objectPositions;
exports.pointerEvents = filters.pointerEvents;
exports.resizes = filters.resizes;
exports.ringBase = filters.ringBase;
exports.rings = filters.rings;
exports.screenReadersAccess = filters.screenReadersAccess;
exports.textOverflows = filters.textOverflows;
exports.textTransforms = filters.textTransforms;
exports.textWraps = filters.textWraps;
exports.transformBase = filters.transformBase;
exports.transforms = filters.transforms;
exports.userSelects = filters.userSelects;
exports.varEmpty = filters.varEmpty;
exports.whitespaces = filters.whitespaces;
exports.writingModes = filters.writingModes;
exports.writingOrientations = filters.writingOrientations;
exports.questionMark = questionMark;
